src/lib/commandBarConfigs/settingsCommandConfig.ts

```diff
@@ -129,10 +125,16 @@ export function createSettingsCommand({
     icon: 'settings',
     needsReview: false,
     onSubmit: (data) => {
-      if (data !== undefined && data !== null) {
-        send({ type: `set.${type}`, data })
+      if (
+        data !== undefined &&
+        data !== null &&
+        'value' in data &&
+        'level' in data
+      ) {
+        const coercedData = data as unknown as SetEventTypes['data']
```

Comment:
I'm guessing the type system was too much to deal with here

Reply:
Yeah it was not happy with the type gymnastics I was attempting. I can add a comment